### PR TITLE
fix: add cmd line mysql ssl options, fixes RHOAIENG-3337

### PR DESCRIPTION
**Pull request by dhirajsb**
## Description

Adds command line options to configure mysql ssl connection.

## How Has This Been Tested?

Verifies that mlmd tries to connect to mysql using ssl options. 

## Merge criteria:

- [X] The commits are squashed in a cohesive manner and have meaningful messages.
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Review by lampajr — APPROVED** [review]
Build is working and the change makes sense to me, thanks @dhirajsb

File: ml_metadata/metadata_store/metadata_store_server_main.cc [hunk 1]
```diff
@@ -328,6 +355,23 @@ DEFINE_string(mysql_config_user, "",
...
+              "certificate authority (CA) certificate(s).");
+DEFINE_string(mysql_config_sslcapath, "",
+              "This parameter specifies path name of the directory "
+              "that contains trusted SSL CA certificates.");
+DEFINE_string(mysql_config_sslcipher, "",
+              "This parameter specifies the list of permissible ciphers for "
+              "SSL encryption.");
+DEFINE_bool(mysql_config_verify_server_cert, false,
+              "This parameter enables verification of the server certificate "
+              " against the host name used when connecting to the server.");
```
Comment (tarilabs) [inline ml_metadata/metadata_store/metadata_store_server_main.cc:374]:
```suggestion
DEFINE_bool(mysql_config_verify_server_cert, false,
              "When certificates are provided, this parameter enables verification of the server certificate "
              " against the host name used when connecting to the server.");
```

I know we discussed this, just re-reporting for archival purposes.

**Review by tarilabs — APPROVED** [review]
per https://github.com/red-hat-data-services/ml-metadata/pull/3#pullrequestreview-1886527344 I believe would be meaningful as well to have @isinyaaa to chime in :)

**Review by antisaling — APPROVED** [review]
lgtm!

**Pull request merged**